examples/java/src/test/java/dev/selenium/actions_api/WheelTest.java

```diff
@@ -0,0 +1,103 @@
+    }
+
+    @Test
+    public void shouldScrollFromViewportByGivenAmount() {
+        driver.get("https://www.selenium.dev/selenium/web/scrolling_tests/frame_with_nested_scrolling_frame_out_of_view.html");
+        WebElement footer = driver.findElement(By.tagName("footer"));
+        int deltaY = footer.getRect().y;
+
+        new Actions(driver).scrollByAmount(0, deltaY).perform();
+
+        Assertions.assertTrue(inViewport(footer));
+    }
+
+    @Test
+    public void shouldScrollFromElementByGivenAmount() {
+        driver.get("https://www.selenium.dev/selenium/web/scrolling_tests/frame_with_nested_scrolling_frame_out_of_view.html");
+        WebElement iframe = driver.findElement(By.tagName("iframe"));
+        WheelInput.ScrollOrigin scrollOrigin = WheelInput.ScrollOrigin.fromElement(iframe);
+
+        new Actions(driver).scrollFromOrigin(scrollOrigin, 0, 200).perform();
+
+        driver.switchTo().frame(iframe);
+        WebElement checkbox = driver.findElement(By.name("scroll_checkbox"));
+        Assertions.assertTrue(inViewport(checkbox));
+    }
+
+    @Test
+    public void shouldScrollFromElementByGivenAmountWithOffset() {
+        driver.get("https://www.selenium.dev/selenium/web/scrolling_tests/frame_with_nested_scrolling_frame_out_of_view.html");
+        WebElement footer = driver.findElement(By.tagName("footer"));
+        WheelInput.ScrollOrigin scrollOrigin = WheelInput.ScrollOrigin.fromElement(footer, 0, -50);
+
+        new Actions(driver).scrollFromOrigin(scrollOrigin,0, 200).perform();
+
+        WebElement iframe = driver.findElement(By.tagName("iframe"));
+        driver.switchTo().frame(iframe);
+        WebElement checkbox = driver.findElement(By.name("scroll_checkbox"));
+        Assertions.assertTrue(inViewport(checkbox));
+    }
+
+    @Test
+    public void shouldScrollFromViewportByGivenAmountFromOrigin() {
+        driver.get("https://www.selenium.dev/selenium/web/scrolling_tests/frame_with_nested_scrolling_frame.html");
+        WheelInput.ScrollOrigin scrollOrigin = WheelInput.ScrollOrigin.fromViewport(10, 10);
+
+        new Actions(driver).scrollFromOrigin(scrollOrigin, 0, 200).perform();
+
+        WebElement iframe = driver.findElement(By.tagName("iframe"));
+        driver.switchTo().frame(iframe);
+        WebElement checkbox = driver.findElement(By.name("scroll_checkbox"));
+        Assertions.assertTrue(inViewport(checkbox));
+    }
+
+    private boolean inViewport(WebElement element) {
+
+        String script =
+                "for(var e=arguments[0],f=e.offsetTop,t=e.offsetLeft,o=e.offsetWidth,n=e.offsetHeight;\n"
+                        + "e.offsetParent;)f+=(e=e.offsetParent).offsetTop,t+=e.offsetLeft;\n"
+                        + "return f<window.pageYOffset+window.innerHeight&&t<window.pageXOffset+window.innerWidth&&f+n>\n"
+                        + "window.pageYOffset&&t+o>window.pageXOffset";
+
+        return (boolean) ((JavascriptExecutor) driver).executeScript(script, element);
```

Comment:
executeScript will return Boolean as Object, seems no need to convert as boolean?